### PR TITLE
Create Projects & Compute Instance

### PR DESCRIPTION
**Pull request by fede1608**
# GCP deploy implementation

### Done so far: 
 - Project Creation
    - User permissions
    - Api service enable
- Compute instance example
- App Engine + Redis Module
- Cloud Run creation

### CODE REVIEW

File: gcp/appengine/main.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  project = "${var.project_id}"
+  service = "appengine.googleapis.com"
+
+  disable_dependent_services = true
+}
+
+resource "google_app_engine_application" "app" {
+  project     = "${var.project_id}"
+  location_id = "${var.location_id}"
+  depends_on = [
```
Comment (mdesanti) [inline gcp/appengine/main.tf]: Why do we need this? It's not recommended to use it as Terraform should handle the dependencies on it's own.

File: gcp/cloudrun/main.tf [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    namespace = "${var.project_id}"
+  }
+
+  spec {
+    containers {
+      image = "gcr.io/cloudrun/hello"
+    }
+  }
+
+  depends_on = [
```
Comment (mdesanti) [inline gcp/cloudrun/main.tf]: Same here

File: gcp/compute/main.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
+data "google_compute_zones" "available" {}
+
+variable project_id {}
+
+resource "google_project_service" "project" {
+  project = "${var.project_id}"
+  service = "compute.googleapis.com"
+
+  disable_dependent_services = true
```
Comment (mdesanti) [inline gcp/compute/main.tf]: Why do we need this?

**Review by mdesanti — COMMENTED** [review]
Can we add a couple of architecture diagrams to show what we're building?

File: gcp/appengine/main.tf (hunk 1 above)
Comment (fede1608) [reply gcp/appengine/main.tf]: If removed, the creation of this resourse starts at the same time as the api service being enabled, so It fails. Terraform does not see it as a dependency so I had to add it manually

File: gcp/compute/main.tf (hunk 3 above)
Comment (fede1608) [reply gcp/compute/main.tf]: while true, it disables all depedencies that were enabled for this api to work.

File: gcp/cloudrun/main.tf (hunk 2 above)
Comment (fede1608) [reply gcp/cloudrun/main.tf]: If removed, the creation of this resourse starts at the same time as the api service being enabled, so It fails. Terraform does not see it as a dependency so I had to add it manually

**Comment by fede1608** [comment]
Added basic gcp deploy diagram

File: gcp/README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# GCP 
+
+## Setup
+
+A terraform admin project must be created to handle all project creation and deployments. Setup `createTerraformAdminProject.sh` variables and run it to create it. Remember that you need `gcloud` installed and setup too.
```
Comment (mdesanti) [inline gcp/README.md]: Can we add a link to Google's documentation on how to install `gcloud`?

File: gcp/createTerraformAdminProject.sh [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# Source: https://cloud.google.com/community/tutorials/managing-gcp-projects-with-terraform
+
+# YOUR_ORG_ID #$ gcloud organizations list
+export TF_VAR_org_id="YOUR_ORG_ID"
```
Comment (mdesanti) [inline gcp/createTerraformAdminProject.sh]: Can we pass this values as arguments when running the script? If we do that, we don't need to change this file when cloning the repo.

**Review by mdesanti — APPROVED** [review]

**Pull request merged**